processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -113,64 +106,116 @@
   private static final EmittingLogger log = new EmittingLogger(IndexIO.class);
   private static final SerializerUtils serializerUtils = new SerializerUtils();
 
-  private static final ObjectMapper mapper;
+  private final ObjectMapper mapper;
+  private final DefaultIndexIOHandler defaultIndexIOHandler;
+  private final ColumnConfig columnConfig;
 
-  protected static final ColumnConfig columnConfig;
+  @Inject
+  public IndexIO(ObjectMapper mapper, ColumnConfig columnConfig)
+  {
+    this.mapper = Preconditions.checkNotNull(mapper, "null ObjectMapper");
+    this.columnConfig = Preconditions.checkNotNull(columnConfig, "null ColumnConfig");
+    defaultIndexIOHandler = new DefaultIndexIOHandler(mapper);
+    indexLoaders = ImmutableMap.<Integer, IndexLoader>builder()
+                               .put(0, new LegacyIndexLoader(defaultIndexIOHandler, columnConfig))
+                               .put(1, new LegacyIndexLoader(defaultIndexIOHandler, columnConfig))
+                               .put(2, new LegacyIndexLoader(defaultIndexIOHandler, columnConfig))
+                               .put(3, new LegacyIndexLoader(defaultIndexIOHandler, columnConfig))
+                               .put(4, new LegacyIndexLoader(defaultIndexIOHandler, columnConfig))
+                               .put(5, new LegacyIndexLoader(defaultIndexIOHandler, columnConfig))
+                               .put(6, new LegacyIndexLoader(defaultIndexIOHandler, columnConfig))
+                               .put(7, new LegacyIndexLoader(defaultIndexIOHandler, columnConfig))
+                               .put(8, new LegacyIndexLoader(defaultIndexIOHandler, columnConfig))
+                               .put(9, new V9IndexLoader(columnConfig))
+                               .build();
 
-  @Deprecated // specify bitmap type in IndexSpec instead
-  protected static final BitmapSerdeFactory CONFIGURED_BITMAP_SERDE_FACTORY;
 
-  static {
-    final Injector injector = GuiceInjectors.makeStartupInjectorWithModules(
-        ImmutableList.<Module>of(
-            new Module()
-            {
-              @Override
-              public void configure(Binder binder)
-              {
-                ConfigProvider.bind(
-                    binder,
-                    DruidProcessingConfig.class,
-                    ImmutableMap.of("base_path", "druid.processing")
-                );
-                binder.bind(ColumnConfig.class).to(DruidProcessingConfig.class);
+  }
 
-                // this property is deprecated, use IndexSpec instead
-                JsonConfigProvider.bind(binder, "druid.processing.bitmap", BitmapSerdeFactory.class);
-              }
-            }
-        )
-    );
-    mapper = injector.getInstance(ObjectMapper.class);
-    columnConfig = injector.getInstance(ColumnConfig.class);
-    CONFIGURED_BITMAP_SERDE_FACTORY = injector.getInstance(BitmapSerdeFactory.class);
+  public void validateTwoSegments(File dir1, File dir2) throws IOException
```

Comment:
this uses `loadIndex` in IndexIO which is not static anymore. if I keep it inside prior location, DefaultIndexIOHandler then that class can not be declared static. It is simpler to move those.